### PR TITLE
Add feature for user to be able to pick image source

### PR DESCRIPTION
**Pull request by Zikstar**
Issue [226](https://github.com/CanHub/Android-Image-Cropper/issues/226)

## Description:
- Added new data class for the user to specify the source to pick Image from
- Added this class to pickImageContract and cropImageContract
- Handled ripple effect of this change

## Check list for the Code Reviewer:
* [ ] CHANGELOG
* [ ] README
* [ ] Wiki
* [ ] Version Number

### CODE REVIEW

**Comment by Canato** [comment]
@Zikstar don't forget to read [CONTRIBUTING](https://github.com/CanHub/Android-Image-Cropper/blob/main/CONTRIBUTING.md)

We will need changelog updates and a new session on `README` to explain how to use image picker

File: cropper/src/main/java/com/canhub/cropper/CropImage.kt [hunk 1]
```diff
@@ -165,8 +169,10 @@ object CropImage {
         return getPickImageChooserIntent(
             context = context,
             title = context.getString(R.string.pick_image_intent_chooser_title),
-            includeDocuments = false,
-            includeCamera = true
+            options = PickImageContractOptions(
+                includeDocuments = false,
+                includeCamera = true
+            )
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropImage.kt:175]: Why we don't have include gallery?

File: cropper/src/main/java/com/canhub/cropper/CropImage.kt [hunk 2]
```diff
@@ -185,22 +191,26 @@ object CropImage {
...
-            getGalleryIntents(
-                packageManager,
-                Intent.ACTION_GET_CONTENT,
-                includeDocuments
+        if (includeGallery) {
+            allIntents.addAll(
+                getGalleryIntents(
+                    packageManager,
+                    Intent.ACTION_GET_CONTENT,
+                    includeDocuments
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropImage.kt:210]: What if `includeDocuments` is true, but `includeGallery` is false?

File: cropper/src/main/java/com/canhub/cropper/PickImageContractOptions.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * @param includeCamera, and
+ * @param includeDocuments,
+ * to include from a particular source just set its
+ * attribute to true, by default Gallery is always
+ * included and documents excluded, so all the user must
+ * always specify is whether to include camera or not*/
+@Parcelize
+data class PickImageContractOptions @JvmOverloads constructor(
+    val includeGallery: Boolean = true,
+    val includeCamera: Boolean,
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/PickImageContractOptions.kt:19]: Let's make this `true` as default too

File: cropper/src/main/java/com/canhub/cropper/PickImageContractOptions.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.canhub.cropper
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+/**
+ * A data class to specify where the user can
+ * select images from, there are 3 options
+ * @param includeGallery,
+ * @param includeCamera, and
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/PickImageContractOptions.kt:10]: remove `and`

**Review by Canato — CHANGES_REQUESTED** [review]
Nice work so far, some small stuff to change

🎉

File: cropper/src/main/java/com/canhub/cropper/CropImage.kt (hunk 1 above)
Comment (Zikstar) [reply cropper/src/main/java/com/canhub/cropper/CropImage.kt:175]: It's set to true by default on the PickImageContractOptions class, I'm trying to preserve the exact behavior that was there before.

File: cropper/src/main/java/com/canhub/cropper/CropImage.kt (hunk 2 above)
Comment (Zikstar) [reply cropper/src/main/java/com/canhub/cropper/CropImage.kt:210]: Great point, I will need to refactor

**Comment by Zikstar** [comment]
New changes/commit made

File: .idea/deploymentTargetDropDown.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (Zikstar) [reply .idea/deploymentTargetDropDown.xml]: File has been deleted

**Comment by Zikstar** [comment]
> @Zikstar the CI is broke, as soon it pass we can merge =D

How do I fix it? I've already ran ktLintFormat. The error messages are by Ktlint about deprecations in files I didn't touch

**Comment by Canato** [comment]
> How do I fix it? I've already ran ktLintFormat. The error messages are by Ktlint about deprecations in files I didn't touch

Look like the error is a unit test

> : /home/runner/work/Android-Image-Cropper/Android-Image-Cropper/cropper/src/test/java/com/canhub/cropper/ContractTestFragment.kt: (32, 66): Type mismatch: inferred type is Boolean but PickImageContractOptions! was expected
> Task :cropper:compileDebugUnitTestKotlin FAILED

Check the `ContractTestFragment` class

To run the tests locally you can type `./gradlew testDebugUnitTest` in the terminal inside Android Studio

**Pull request merged**